File: pyg-manager-web/src/main/java/com/pyg/manager/controller/BrandController.java
package com.pyg.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pyg.manager.service.BrandService;
import com.pyg.pojo.TbBrand;
import com.pyg.vo.InfoResult;
import com.pyg.vo.PageResult;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * Description:
 * @author AK
 * @date 2018/8/6 20:19
 * @since 1.0.0
 */
@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference(timeout=100000)
    private BrandService brandService;

    /**
     * description: 返回全部列表
     *
     * @return java.util.List<com.pyg.pojo.TbBrand>
     * @author AK
     * @date  2018年08月09日 10:18:22
     */
    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        return brandService.findAll();
    }

    /**
     * description: 分页返回全部列表
     *
     * @param page 当前页面
     * @param rows 每页记录数
     * @return com.pyg.vo.PageResult
     * @author AK
     * @date  2018年08月06日 21:50:03
     */
    @RequestMapping("/findPage/{page}/{rows}")
    public PageResult findPage(@PathVariable Integer page, @PathVariable Integer rows) {
        return brandService.findPage(page, rows);
    }

    /**
     * description: 增加
     *
     * @param brand 实体
     * @return com.pyg.vo.InfoResult
     * @author AK
     * @date  2018年08月06日 21:54:13
     */
    @RequestMapping("/add")
    public InfoResult add(@RequestBody TbBrand brand) {
        try {
            brandService.add(brand);
            return new InfoResult(true, "增加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new InfoResult(false, "增加失败");
        }
    }

    /**
     * description: 根据id返回实体
     *
     * @param id 实体id
     * @return com.pyg.pojo.TbBrand
     * @author AK
     * @date  2018年08月06日 21:55:22
     */
    @RequestMapping("/findOne/{id}")
    public TbBrand findOne(@PathVariable Long id) {
        return brandService.findOne(id);
    }

    /**
     * description: 修改
     *
     * @param brand 实体
     * @return com.pyg.vo.InfoResult
     * @author AK
     * @date  2018年08月09日 10:22:38
     */
    @RequestMapping("/update")
    public InfoResult update(@RequestBody TbBrand brand) {
        try {
            brandService.update(brand);
            return new InfoResult(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new InfoResult(false,"修改失败");
        }
    }

    /**
     * description: 批量删除
     *
     * @param ids 实体ids
     * @return com.pyg.vo.InfoResult
     * @author AK
     * @date  2018年08月07日 08:25:09
     */
    @RequestMapping("/delete/{ids}")
    public InfoResult delete(@PathVariable Long[] ids) {
        try {
            brandService.delete(ids);
            return new InfoResult(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new InfoResult(false, "删除失败");
        }
    }

    /**
     * description: 查询+分页
     *
     * @param page 当前页
     * @param rows 每页记录数
     * @param brand 查询条件
     * @return com.pyg.vo.PageResult
     * @author AK
     * @date  2018年08月07日 19:56:59
     */
    @RequestMapping("/search/{page}/{rows}")
    public PageResult search(@PathVariable Integer page, @PathVariable Integer rows, @RequestBody TbBrand brand) {
        return brandService.findPage(page, rows, brand);
    }

    @RequestMapping("/findBrandList")
    public List<Map> findBrandList() {
        return brandService.selectOptionList();
    }
}

File: pyg-manager-web/src/main/webapp/js/controller/brandController.js
// 控制层
app.controller('brandController', function ($scope, $controller, brandService) {
    // 继承baseController
    $controller('baseController', {$scope: $scope}); // 继承

    // 读取列表数据绑定到表单中
    $scope.findAll = function () {
        brandService.findAll().success(function (response) {
            $scope.list = response;
        })
    };

    // 分页
    $scope.findPage = function(page, rows) {
        brandService.findPage(page, rows).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;
        })
    };

    // 保存
    $scope.save = function () {
        var object = brandService.add($scope.entity);
        if($scope.entity.id != null) {
            object = brandService.update($scope.entity);
        }
        object.success(function (response) {
            if(response.success) {
                $scope.reloadList();
            } else {
                alert(response.message);
            }
        })
    };

    // 查询实体
    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity = response;
        })
    };

    // 批量删除
    $scope.del = function () {
        // 获取选中的复选框
        brandService.del($scope.selectIds).success(function (response) {
            if(response.success) {
                // 清空删除ids
                $scope.selectIds = [];
                $scope.reloadList(); // 刷新列表
            } else {
                alert(response.message);
            }
        })
    };
    
    $scope.searchEntity = {}; // 定义搜索对象
    // 搜索
    $scope.search = function (page, rows) {
        brandService.search(page, rows, $scope.searchEntity).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total; // 更新总记录数
        })
    };

});